connector/spanmetricsconnector/connector.go

```diff
@@ -115,17 +117,23 @@ func newConnector(logger *zap.Logger, config component.Config, ticker *clock.Tic
 		return nil, err
 	}
 
+	resourceMetricsKeyAttributes := make(map[string]bool)
+	for i := 0; i < len(cfg.ResourceMetricsKeyAttributes); i++ {
+		resourceMetricsKeyAttributes[cfg.ResourceMetricsKeyAttributes[i]] = true
+	}
```

Comment:
nit: we can give a capacity hint since we know the final size of the map at the completion of the `for` loop.
Also, it's more idiomatic in Go to use `range` if we just want to iterate over elements of a slice. It has the added benefit of being _a little_ more succinct in this case. 😄 

```suggestion
	resourceMetricsKeyAttributes := make(map[string]bool, len(cfg.ResourceMetricsKeyAttributes))
	for _, attr := range cfg.ResourceMetricsKeyAttributes {
		resourceMetricsKeyAttributes[attr] = true
	}
```